src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableQueue.cs

```diff
@@ -34,47 +35,69 @@ public static ImmutableQueue<T> Create<T>()
         [Pure]
         public static ImmutableQueue<T> Create<T>(T item)
         {
-            return ImmutableQueue<T>.Empty.Enqueue(item);
+            return new ImmutableQueue<T>(ImmutableStack.Create(item), ImmutableStack<T>.Empty);
```

Comment:
Hm, it looks like this is causing `Create` to not be inlined causing the perf difference to be nil.